.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,119 @@
+name: QMI Python CI runner
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  # minimum pylint code quality score (max. 10)
+  PYLINT_MIN_SCORE: "9.00"
+  # minimum code coverage (goal: 80%; max. 100%)
+  COVERAGE_MIN_PERC: "68"
+  # maximum code complexity (goal: <= 15; unbounded)
+  COMPLEXITY_MAX_SCORE: "38"
+  # minimum maintainability index (goal: >=20; max. 100)
+  MAINTAINABILITY_MIN_INDEX: "10.62"
+  SOURCE_DIRS: "qmi/"
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python-latest
+        uses: actions/setup-python@v4
+        with:
+          # Take latest Python 3 version
+          python-version: "3.x"
```

Comment:
currently QMI is officially supported for Python 3.8 onwards so we should also run these steps with the older versions

Reply:
I was thinking of leaving that for the "daily" checker only in https://github.com/QuTech-Delft/QMI/blob/6-add-ci-workflows-into-the-github-directory/.github/workflows/schedule-full-ci-test-main.yml. Like we have now. This to reduce the total run-time of a simple "push" action.

Reply:
this pipeline also runs for a PR, so we should check for the different versions before merging, especially since we see outside contributors making PRs in the future. Also the master branch should be protected so now one can push to it directly

Reply:
OK, done. Added a separate workflow file for pull requests.